pageserver/src/repository.rs

```diff
@@ -119,24 +123,30 @@ pub trait Timeline: Send + Sync {
     fn gc_iteration(&self, horizon: u64) -> Result<GcResult>;
 }
 
-pub trait History: Iterator<Item = Result<RelationUpdate>> {
+pub trait History: Iterator<Item = Result<Modification>> {
     /// The last_valid_lsn at the time of history() call.
     fn lsn(&self) -> Lsn;
 }
 
+//
+// Structure representing any update operation of object storage.
+// It is used to copy object storage content in PUSH method.
+//
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct RelationUpdate {
-    pub rel: RelTag,
+pub struct Modification {
+    pub tag: ObjectTag,
     pub lsn: Lsn,
-    pub update: Update,
+    pub data: Vec<u8>,
```

Comment:
If the intent is just to ship key/values without thinking much about them, I think it's fine to store the value as serialized. Certainly a departure from the earlier design though

Reply:
We have object (actually KV pairs) in one object storage and what to copy them to another storage. So we just need to pass this KV pairs through network in some portable format with as less transformations as possible. This was the intent. Why do need push logic to be concerned about semantic of this KV pairs?